src/MissionManager/ComplexMissionItem.cc

```diff
@@ -396,6 +396,24 @@ void ComplexMissionItem::_intersectLinesWithPolygon(const QList<QLineF>& lineLis
     }
 }
 
+/// Adjust the line segments such that they are all going the same direction with respect to going from P1->P2
+void ComplexMissionItem::_adjustLineDirection(const QList<QLineF>& lineList, QList<QLineF>& resultLines)
+{
+    for (int i=0; i<lineList.count(); i++) {
+        const QLineF& line = lineList[i];
+        QLineF adjustedLine;
+
+        if (line.angle() > 180.0) {
```

Comment:
That's not really what this code does.  It's just ordering the endpoints of the parallel grid lines (P1, P2) such that the vectors of the lines are all pointing in the same direction. The lines themselves stay the same, only the endpoints are flipped. This way when you connect the turn around from the end of one parallel line to the beginning of the next the connection is always on the edge of the polygon instead of possibly traversing the center of the polygon as shown in the image.

I think it's correct, unless I'm misunderstanding your point.